components/Leaderboard/MintNft/utils.jsx

```diff
@@ -3,26 +3,29 @@ import { notifyError, notifySuccess } from '@autonolas/frontend-library';
 
 import { getMintContract } from 'common-util/Contracts';
 import { GATEWAY_URL } from 'util/constants';
+import { getEstimatedGasLimit } from 'common-util/functions/requests';
 
 const pattern = /ipfs:\/\/+/g;
 export const getAutonolasTokenUri = (tokenUri) => (tokenUri || '').replace(pattern, GATEWAY_URL);
 
 export const mintNft = (account) => new Promise((resolve, reject) => {
   const contract = getMintContract();
 
-  contract.methods
-    .mint()
-    .send({ from: account })
-    .then((response) => {
-      notifySuccess('Successfully Minted');
-      const id = get(response, 'events.Transfer.returnValues.id');
-      resolve(id);
-    })
-    .catch((e) => {
-      notifyError('Error: could not mint NFT');
-      window.console.log('Error occured on minting NFT');
-      reject(e);
-    });
+  const mintFn = contract.methods.mint();
+
+  getEstimatedGasLimit(mintFn, account).then((estimatedGas) => {
+    mintFn.send({ from: account, gasLimit: estimatedGas })
```

Comment:
pls check `common-util/functions/requests.jsx` we use Math.floor() on the gas estimate, which is a little strange

perhaps we should add a small multiplier to the estimate too? i.e.
```suggestion
    mintFn.send({ from: account, gasLimit: estimatedGas * 1.1 })
```

Reply:
I think we should use `estimatedGas` directly without `Math.floor`, it was copy-pasted from registry app as I remember. Maybe in old ethers versions the result was not an integer value so that's why it was rounded down? @mohandast52 do you remember by any chance?

Reply:
also @Tanya-atatakai getMintContract returns a web3 contract,
should use `gas` instead of `gasLimit`
https://web3js.readthedocs.io/en/v1.2.11/web3-eth-contract.html#id33

Reply:
updated rounding in `common-util/functions/requests.jsx` to `Math.floor(estimatedGas * 1.1)` so the result is an integer

Reply:
@kupermind can you please review this PR as well just in case